cloudconfig/cloudinit/network_ubuntu.go

```diff
@@ -260,8 +260,8 @@ func PrepareNetworkConfigFromInterfaces(interfaces corenetwork.InterfaceInfos) (
 		}
 		if cidr != "" {
 			nameToAddress[ifaceName] = cidr
-		} else if info.ConfigType == corenetwork.ConfigDHCP {
-			nameToAddress[ifaceName] = string(corenetwork.ConfigDHCP)
+		} else if info.ConfigMethod == network.DynamicAddress {
```

Comment:
Here is the reason we needed a provisioner API bump for backward compatibility.

The only place that backward compatibility was broken was for container interface info sent by the provisioner to local container brokers. Those would be expecting "dhcp" instead or "dynamic".